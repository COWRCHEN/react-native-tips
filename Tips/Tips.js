import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import ReactMixin from 'react-mixin'
import TimerMixin from 'react-timer-mixin'
import styled from 'styled-components/native'
import {
  Text, View, Modal, StyleSheet,
  TouchableOpacity, ViewPropTypes
} from 'react-native'


const getArrowStyleByPosition = (position = 'top') => {
  switch (position) {
    case 'right': return `
      left: -5px;
      top: 50%;
      margin-top: 5px;
      border-top-width: 5px;
      border-top-color: transparent;
      border-bottom-width: 5px;
      border-bottom-color: transparent;
      border-right-width: 5px;
      border-right-color: rgba(0, 0, 0, 0.5);
    `

    case 'left': return `
      right: -5px;
      top: 50%;
      margin-top: 5px;
      border-top-width: 5px;
      border-top-color: transparent;
      border-bottom-width: 5px;
      border-bottom-color: transparent;
      border-left-width: 5px;
      border-left-color: rgba(0, 0, 0, 0.5);
    `

    case 'bottom': return `
      top: -5px;
      left: 50%;
      margin-left: 2.5px;
      border-bottom-width: 5px;
      border-bottom-color: rgba(0, 0, 0, 0.5);
      border-right-width: 5px;
      border-right-color: transparent;
      border-left-width: 5px;
      border-left-color: transparent;
    `

    default: return `
      bottom: -5px;
      left: 50%;
      margin-left: 2.5px;
      border-top-width: 5px;
      border-top-color: rgba(0, 0, 0, 0.5);
      border-right-width: 5px;
      border-right-color: transparent;
      border-left-width: 5px;
      border-left-color: transparent;
    `
  }
}

const TooltipArrow = styled.View`
  position: absolute;
  shadow-offset: { width: 0, height: 0 };
  shadow-radius: 1px;
  shadow-color: black;
  shadow-opacity: 0.5;
  ${props => getArrowStyleByPosition(props.position)}
`

const Tooltip = styled.View`
  background: rgba(0, 0, 0, 0.5);
  flex: -1;
  padding: 10px;
  border-radius: 4px;
  shadow-offset: { width: 0, height: 0 };
  shadow-radius: 1px;
  shadow-color: black;
  shadow-opacity: 0.5;
`

const ModalContent = styled.View`
  width: 80%;
  display: flex;
  align-self: center;
  position: absolute;
  flex-wrap: nowrap;
`

const ChildrenOverlay = styled.TouchableOpacity`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`


const styles = StyleSheet.create({
  tooltipContainer: {
    position: 'relative',
    flex: 1
  },

  modal: {
    backgroundColor: 'rgba(0, 0, 0, 0.6)',
    position: 'relative',
    flex: 1
  },

  text: {
    color: 'white',
    textAlign: 'center'
  }
})


export default class Tips extends PureComponent {
  /* LIFECYCLE */

  /**
   * @constructor
   * @param {*} props - Properties of the PureComponent
   */
  constructor(props) {
    super(props)

    this.state = {

      /**
       * The position of the children in x axis
       * @type {Number}
       */
      componentLeft: 0,

      /**
       * The position of the children in y axis
       * @type {Number}
       */
      componentTop: 0,

      /**
       * The width of the children
       * @type {Number}
       */
      componentWidth: 0,

      /**
       * The height of the children
       * @type {Number}
       */
      componentHeight: 0,

      /**
       * The position of the tooltip in x axis
       * @type {Number}
       */
      tooltipLeft: 0,

      /**
       * The position of the tooltip in y axis
       * @type {Number}
       */
      tooltipTop: 0,

      /**
       * Check if the component is ready
       * @type {Boolean}
       */
      ready: false
    }

    this.view = null

    this.handleLayout = this.handleLayout.bind(this)
    this.handleTooltipLayout = this.handleTooltipLayout.bind(this)
    this.handleRequestNext = this.handleRequestNext.bind(this)
  }

  /**
   * @componentWillReceiveProps
   * @param {*} nextProps - Next properties of the PureComponent
   */
  componentWillReceiveProps(nextProps, nextState) {
    if (nextProps.visible && !this.props.visible && !nextState.ready) {
      this.updateComponentPosition(true)
    } else if (!nextProps.visible && this.props.visible && nextState.ready) {
      this.setState({
        ready: false
      })
    }
  }


  /* METHODS */

  /**
   * Extends the stylesheet of the tooltip container to be relative to the
   * position passed in parameters
   * @param {String} position - The position
   * @returns {Object} Stylesheet properties
   */
  getModalContentStyleByPosition(position) {
    switch (position) {
      case 'right':
      case 'left': return {
        flexDirection: 'row'
      }

      case 'top':
      case 'bottom': return {
        flexDirection: 'column'
      }

      default: return {}
    }
  }

  /**
   * Call the measure to ensure positionning
   * @param {Boolean=} willBeReady - If true, the component state will be ready to be visible
   */
  updateComponentPosition(willBeReady = false) {
    this.requestAnimationFrame(() => {
      this.view.measure((x, y, width, height, pageX, pageY) => {
        this.setState(state => ({
          ready: willBeReady || state.ready,
          componentLeft: pageX,
          componentTop: pageY,
          componentWidth: state.componentWidth || width,
          componentHeight: state.componentHeight || height
        }))
      })
    })
  }

  /**
   * Measure the size of the children
   * @param {*} event - The event object
   */
  handleLayout(event) {
    const { width, height } = event.nativeEvent.layout

    this.setState({
      componentWidth: width,
      componentHeight: height
    })

    this.updateComponentPosition()
  }

  /**
   * Measure the size of the tooltip layout
   * @param {*} event - The event object
   */
  handleTooltipLayout(event) {
    const { position } = this.props
    const { width, height } = event.nativeEvent.layout

    this.setState((state) => {
      const {
        componentLeft, componentTop, componentWidth, componentHeight
      } = state
      const nextState = {}

      switch (position) {
        case 'right':
          nextState.tooltipTop = Math.max(-componentTop, (componentHeight / 2) - (height / 2))
          nextState.tooltipLeft = 10
          break

        case 'left':
          nextState.tooltipTop = Math.max(-componentTop, (componentHeight / 2) - (height / 2))
          nextState.tooltipLeft = -componentLeft - width - 10
          break

        case 'bottom':
          nextState.tooltipLeft = Math.max(-componentLeft, (componentWidth / 2) - (width / 2))
          nextState.tooltipTop = 10
          break

        case 'top':
          nextState.tooltipTop = Math.max(-componentTop, -componentHeight - height - 10)
          nextState.tooltipLeft = Math.max(-componentLeft, (componentWidth / 2) - (width / 2))
          break

        default:
          nextState.tooltipTop = 0
          nextState.tooltipLeft = 0
          break
      }

      return nextState
    })
  }

  /**
   * Handle event to switch between 'onRequestNext' and 'onRequestClose'
   */
  handleRequestNext() {
    const { onRequestNext, onRequestClose } = this.props

    if (onRequestNext) {
      return onRequestNext()
    }

    return !!onRequestClose && onRequestClose()
  }


  /* RENDER */

  /**
   * @render
   * @returns {JSX}
   */
  render() {
    const {
      children, position, text,
      modalStyle, textStyle, style, contentStyle,
      content, offsetLeft, offsetTop
    } = this.props

    const {
      componentLeft, componentTop,
      componentWidth: width, componentHeight: height,
      tooltipLeft, tooltipTop, ready
    } = this.state

    const left = componentLeft + offsetLeft
    const top = componentTop + offsetTop
    const visible = this.props.visible && ready

    return (
      <View
        collapsable={false}
        renderToHardwareTextureAndroid
        ref={(view) => { this.view = view }}
      >
        <View onLayout={this.handleLayout}>{children}</View>

        <Modal
          animationType="fade"
          visible={visible}
          transparent
        >
          <TouchableOpacity
            activeOpacity={1}
            focusedOpacity={1}
            style={{ flex: 1 }}
            onPress={this.handleRequestNext}
          >
            <View style={[styles.modal, modalStyle]}>
              <ModalContent
                style={[
                  this.getModalContentStyleByPosition(position),
                  { left, top },
                  contentStyle
                ]}
              >
                <View style={{ width, height }}>
                  {children}
                </View>

                <View
                  onLayout={this.handleTooltipLayout}
                  style={[styles.tooltipContainer, {
                    top: tooltipTop,
                    left: tooltipLeft
                  }]}
                >
                  <Tooltip
                    style={style}
                  >
                    {content}
                    {text && <Text style={[styles.text, textStyle]}>{text}</Text>}
                    {position !== 'none' && <TooltipArrow position={position} />}
                  </Tooltip>
                </View>

                <ChildrenOverlay
                  onPress={this.handleRequestNext}
                  style={styles.childrenOverlay}
                />
              </ModalContent>

            </View>
          </TouchableOpacity>
        </Modal>
      </View>
    )
  }
}


Tips.defaultProps = {
  visible: false,
  style: {},
  textStyle: {},
  modalStyle: {},
  contentStyle: {},
  children: [],
  content: [],
  offsetLeft: 0,
  offsetTop: 0,
  text: '',
  position: 'top',
  onRequestClose: () => {},
  onRequestNext: () => {}
}


Tips.propTypes = {
  position: PropTypes.oneOf(['left', 'top', 'bottom', 'right', 'none']),
  style: ViewPropTypes.style,
  modalStyle: ViewPropTypes.style,
  contentStyle: ViewPropTypes.style,
  textStyle: ViewPropTypes.style,
  offsetLeft: PropTypes.number,
  offsetTop: PropTypes.number,
  onRequestClose: PropTypes.func,
  onRequestNext: PropTypes.func,
  children: PropTypes.node,
  text: PropTypes.string,
  content: PropTypes.node,
  visible: PropTypes.bool
}


ReactMixin(Tips.prototype, TimerMixin)
