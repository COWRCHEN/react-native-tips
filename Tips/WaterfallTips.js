export default class WaterfallTips {
  /**
   * @constructor
   * @param {*} indexes - Indexes to create
   */
  constructor(indexes = []) {
    this.index = null
    this.indexes = {}

    indexes.forEach(index => this.createIndex(index))
  }

  /**
   * Start the waterfall tips by the first position
   */
  start() {
    this.index = this.getIndexByPosition(0)

    return this.index && this.index.name
  }

  /**
   * Check if an alement must be visible or not
   * @param {String|Object} index - The index
   * @returns {Boolean} Is visible or not
   */
  isVisible(index = '') {
    return index === this.index.name
  }

  /**
   * Create a new index
   * @param {String} name - The name of the index
   * @returns {*} The index object
   */
  createIndex(name) {
    if (!name) {
      throw new Error(`WaterfallTips.createIndex must have a name parameter. Found: "${name}"`)
    }

    const key = name.replace(/[^a-zA-Z]/g, '')

    if (this.indexes[key]) {
      throw new Error(`WaterfallTips index must be unique. Found duplicate key: "${key} (${name})"`)
    }

    const index = {
      name: key,
      position: Object.keys(this.indexes).length
    }

    this.indexes[key] = index
    return key
  }

  /**
   * Get the index by its position
   * @param {Number} position - The position of the index
   * @returns {*} The index object
   */
  getIndexByPosition(position) {
    const key = Object.keys(this.indexes)
      .find(k => this.indexes[k].position === position)

    return this.indexes[key]
  }


  /* EVENTS */

  /**
   * Trigger the next tips
   */
  next() {
    if (!this.index
      || (this.index && this.index.position + 1 >= Object.keys(this.indexes).length)) {
      this.index = null
    } else {
      this.index = this.getIndexByPosition(this.index.position + 1)
    }

    return this.index && this.index.name
  }

  /**
   * Trigger the previous tips
   */
  previous() {
    if (this.index && this.index.position > 0) {
      this.index = this.getIndexByPosition(this.index.position - 1)
    }

    return this.index && this.index.name
  }
}
