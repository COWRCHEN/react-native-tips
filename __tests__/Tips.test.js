import React from 'react'
import { Text } from 'react-native'

import Tips from './../index'

import renderer from 'react-test-renderer';


describe('<Tips />', () => {
  it('should render and not be visible', () => {
    const tree = renderer.create((
      <Tips>
        <Text>Highlighted text !</Text>
      </Tips>
    )).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should make tips visible', () => {
    const tree = renderer.create((
      <Tips
        visible
      >
        <Text>Highlighted text !</Text>
      </Tips>
    )).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
