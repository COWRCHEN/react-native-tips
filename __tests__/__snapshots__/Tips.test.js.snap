// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tips /> should make tips visible 1`] = `
<View
  collapsable={false}
  renderToHardwareTextureAndroid={true}
>
  <View
    onLayout={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Highlighted text !
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.6)",
              "flex": 1,
              "position": "relative",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "display": "flex",
                "flexWrap": "nowrap",
                "position": "absolute",
                "width": "80%",
              },
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 0,
                  "width": 0,
                },
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Highlighted text !
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "position": "relative",
                },
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                    "borderRadius": 4,
                    "flexBasis": 0,
                    "flexGrow": -1,
                    "flexShrink": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                    "shadowColor": "black",
                    "shadowOpacity": 0.5,
                    "shadowRadius": 1,
                  },
                  Object {},
                ]
              }
            >
              <View
                position="top"
                style={
                  Array [
                    Object {
                      "borderLeftColor": "transparent",
                      "borderLeftWidth": 5,
                      "borderRightColor": "transparent",
                      "borderRightWidth": 5,
                      "borderTopColor": "rgba(0, 0, 0, 0.5)",
                      "borderTopWidth": 5,
                      "bottom": -5,
                      "left": "50%",
                      "marginLeft": 2.5,
                      "position": "absolute",
                      "shadowColor": "black",
                      "shadowOpacity": 0.5,
                      "shadowRadius": 1,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`<Tips /> should render and not be visible 1`] = `
<View
  collapsable={false}
  renderToHardwareTextureAndroid={true}
>
  <View
    onLayout={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Highlighted text !
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.6)",
              "flex": 1,
              "position": "relative",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "display": "flex",
                "flexWrap": "nowrap",
                "position": "absolute",
                "width": "80%",
              },
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 0,
                  "width": 0,
                },
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Highlighted text !
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "position": "relative",
                },
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                    "borderRadius": 4,
                    "flexBasis": 0,
                    "flexGrow": -1,
                    "flexShrink": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                    "shadowColor": "black",
                    "shadowOpacity": 0.5,
                    "shadowRadius": 1,
                  },
                  Object {},
                ]
              }
            >
              <View
                position="top"
                style={
                  Array [
                    Object {
                      "borderLeftColor": "transparent",
                      "borderLeftWidth": 5,
                      "borderRightColor": "transparent",
                      "borderRightWidth": 5,
                      "borderTopColor": "rgba(0, 0, 0, 0.5)",
                      "borderTopWidth": 5,
                      "bottom": -5,
                      "left": "50%",
                      "marginLeft": 2.5,
                      "position": "absolute",
                      "shadowColor": "black",
                      "shadowOpacity": 0.5,
                      "shadowRadius": 1,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`<Tips /> should render tips without children 1`] = `
<View
  collapsable={false}
  renderToHardwareTextureAndroid={true}
>
  <View
    onLayout={[Function]}
  />
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.6)",
              "flex": 1,
              "position": "relative",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "display": "flex",
                "flexWrap": "nowrap",
                "position": "absolute",
                "width": "80%",
              },
              Array [
                Object {},
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 0,
                  "width": 0,
                },
                Object {},
              ]
            }
          />
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "position": "relative",
                },
                Object {
                  "left": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                    "borderRadius": 4,
                    "flexBasis": 0,
                    "flexGrow": -1,
                    "flexShrink": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                    "shadowColor": "black",
                    "shadowOpacity": 0.5,
                    "shadowRadius": 1,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
